### DISCUSSION TITLE
Duplicating flexible editor content including relational nodes with flow

### DISCUSSION THREAD

**Discussion by jdjoshuadavison** [formfcw/directus-extension-flexible-editor#24] [Q&A] [answered]
Hello, I have a flow that duplicates a content item. Presumably I will need to duplicate every relation node in my junction table as well, update the IDs in the duplicated relation node items, and update the IDs in the flexible editor JSON. Is that right - there isn't an easier way? Is there anything else I'll need to think about?

**Comment by formfcw** [comment, marked as answer]
Thanks for the question. I agree that duplication is not quite convenient due to the way it works in directus and the complexity with referencing M2A items in a separate JSON field.

I'm afraid that setting the `Item Duplication Fields` for each content collection as well is the only way to get this to work. 

However, to workaround this inconvenience, you can go to your editor nodes junction collection (like `post_editor_nodes`), `Copy Raw Value` from the already set `Item Duplication Fields`, paste it into the `Item Duplication Fields` of your content collection (like `posts`) and prefix all copied field values with the M2A field name of your content collection.

For example, copy this from the `Item Duplication Fields` of `post_editor_nodes`:

```json
[
	"collection",
	"item:related_content.items.collection",
	"item:related_content.items.sort",
	"item:related_content.items.item:pages.id",
	"item:related_content.items.item:posts.id",
	"item:gallery.items.sort",
	"item:gallery.items.directus_files_id",
	"item:video.video_url",
	"item:video.video_poster"
]
```

And paste the fields prefixed into the `Item Duplication Fields` in `posts`:

```json
[
	"…"
	"editor_nodes.collection",
	"editor_nodes.item:related_content.items.collection",
	"editor_nodes.item:related_content.items.sort",
	"editor_nodes.item:related_content.items.item:pages.id",
	"editor_nodes.item:related_content.items.item:posts.id",
	"editor_nodes.item:gallery.items.sort",
	"editor_nodes.item:gallery.items.directus_files_id",
	"editor_nodes.item:video.video_url",
	"editor_nodes.item:video.video_poster"
]
```

Even if the content items are duplicated, the duplicated M2A junction items loose the connection to the referenced IDs in the Editor JSON. So this issue still exists, but should be solved with [rewriting the whole duplication process](https://github.com/formfcw/directus-extension-flexible-editor/discussions/11).
